cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/NetworkConnector.java

```diff
@@ -1,26 +1,99 @@
 package com.sequenceiq.cloudbreak.cloud;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
 import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
 import com.sequenceiq.cloudbreak.cloud.model.Network;
 import com.sequenceiq.cloudbreak.cloud.model.SubnetSelectionParameters;
+import com.sequenceiq.cloudbreak.cloud.model.SubnetSelectionResult;
 import com.sequenceiq.cloudbreak.cloud.model.network.CreatedCloudNetwork;
 import com.sequenceiq.cloudbreak.cloud.model.network.NetworkCreationRequest;
 import com.sequenceiq.cloudbreak.cloud.model.network.NetworkDeletionRequest;
-import com.sequenceiq.cloudbreak.cloud.model.SubnetSelectionResult;
 
 /**
  * Network connectors.
  */
 public interface NetworkConnector extends CloudPlatformAware {
+    Logger LOGGER = LoggerFactory.getLogger(NetworkConnector.class);
+
+    String NOT_ENOUGH_AZ = "Acceptable subnets are in %d different AZs, but subnets in %d different AZs required.";
 
     CreatedCloudNetwork createNetworkWithSubnets(NetworkCreationRequest networkCreationRequest);
 
     void deleteNetworkWithSubnets(NetworkDeletionRequest networkDeletionRequest);
 
     String getNetworkCidr(Network network, CloudCredential credential);
 
-    SubnetSelectionResult selectSubnets(List<CloudSubnet> subnetMetas, SubnetSelectionParameters subnetSelectionParameters);
+    SubnetSelectionResult filterSubnets(Collection<CloudSubnet> subnetMetas, SubnetSelectionParameters subnetSelectionParameters);
```

Comment:
The result of filterSubnets is always used in conjunction with select (except for logging). I therefore would not put filterSubnets to the api but would make it a private implementation within AWS / Azure / Mock.